fix: ECS Fargate enhancemente and exec doc (#1092)
appsmithorg/appsmith-docs: website/docs/getting-started/setup/installation-guides/aws-ecs-on-fargate.md

    Ensure that the EFS, ECS cluster, and Fargate instances are all in the same VPC.
    :::
      
    3. Click the EFS created and navigate to the Network tab to ensure the mount-target is made in the same availability zone as the ECS cluster.
    4. Create a new security group to allow inbound and outbound NFS traffic.
    5. Attach the security group to the EFS mount-target.

> 5. **Create the ECS task role:**

Reviewer: Let's keep it as Sumesh suggested 'task execution role' . Just read up that is some difference between both

Reviewer 2: I've changed this. @sum35h - Please confirm.

Reviewer: Oh sorry, my bad. I might have been confused because in some parts of the doc, I saw task execution role.

Author: This is a change to what was earlit task execution role. Now it's just task role

>     To enable the `exec` command, follow the steps below:

Reviewer: ```suggestion
```

Reviewer 2: This is needed as suggested by Sumesh.

Author: Earlier, I wanted to highlight this as a note.

Reviewer 2: The callout was not needed as it's information where the steps allow users to enable the `exec` command.

    1. Go to the **IAM** console and select **Roles**.
    2. Click Create Role.
    3. Select trust entity:
        1. Select the Trusted entity type as AWS Service.
        2. Select **Elastic Container Service Task** as the use case, and hit next.

    4. Add permission
        1. Add `SecretsManagerReadWrite`.
        2. Click Create Policy to open the policy editor, and choose JSON mode.

        ```json title="Custom IAM Policy"
        {
            "Version": "2012-10-17",
            "Statement": [
                {
                    "Effect": "Allow",
                    "Action": [
                        "ssmmessages:CreateControlChannel",
                        "ssmmessages:CreateDataChannel",
                        "ssmmessages:OpenControlChannel",
                        "ssmmessages:OpenDataChannel"
                    ],
                    "Resource": "*"
                }
                ]
        }

        ```

>     5. Name, Review, and Create:

Reviewer: Can we directly add the step here as this heading takes away focus from what they actually have to do

Reviewer 2: @sum35h

>     Name the role and review that the trust policy has `ecs-tasks.amazonaws.com` and `sts:AssumeRole` permissions as shown below:

Reviewer: there is some confusion here. `ecs-tasks.amazonaws.com` is not a permission. Let reword this a bit.

```suggestion
Name the role and ensure that the IAM task execution role has a trust policy that specifies the `ecs-tasks.amazonaws.com` service with the `sts:AssumeRole` permission as shown below:       
```

Reviewer: I'd say it's okay if the readability on the vale linter breaks, but we should not accidentally change the meaning.

Reviewer 2: Based on my discussion with Sumesh, the trust policy already has these details. We want users to review if these permissions are present. We rewrote it together. @sum35h

Reviewer: Ok but still ecs-tasks.amazonaws.com is not a permission right? but in the statement it comes as that

Reviewer: @jnikhila can we fix this one?

Something like

> Name the role and ensure that the  trust policy specifies the `ecs-tasks.amazonaws.com` service with the `sts:AssumeRole` permission as shown below:

    ```json title="Trust Policy"
    {
        "Version": "2008-10-17",
        "Statement": [
            {
                "Sid": "",
                "Effect": "Allow",
                "Principal": {
                    "Service": "ecs-tasks.amazonaws.com"
                },
                "Action": "sts:AssumeRole"
            }
        ]
    
    }

    ```

>     Once done, click **Create**.

Reviewer: so this looks like part of the steps

```suggestion
    6.  Click **Create**.
```

1. On the sidebar, choose Task Definitions and Create a new Task Definition.
2. Choose Fargate as the launch type, and proceed to the next step.
3. Enter the task definition name.

> 4. Set the task role to the one created in prerequisites - **Create the ECS task role**.

Reviewer: ```suggestion
4. Set the task role to the one created in step 5 **Create the ECS task execution role** under **Prerequisites**.
```

5. Select the default Network mode.
6. Set Linux as the Operating system family.
7. Set the Task Execution Role option to [**Create new role**].
8. Set the required task size for memory & CPU (Minimum requirement: 2vCPU and 4 GB Memory).
9. Go to the **Volumes** section and add a new volume. Enter the Name as `appsmith_stack`, set the Volume type as **EFS**, and set the **File System ID** to the EFS filesystem created in the prerequisite step. Leave the remaining fields with the default values.

    ![EFS Volume Configuration](/img/ecs-efs-fargate.png)

> 10. Configure the **Appsmith container**.

Reviewer: ```suggestion
10. Configure the Appsmith container.
```

    1. Click the **Add container** button.
    2. Enter the container name, and set the Image to `appsmith/appsmith-ce`.
    3. Add port mappings for ports **80->80,443->443**.
    4. Set the _Mount points Source volume_ to `appsmith_stack` and set the Container path to `/appsmith-stacks`.

    ![Storage Setting](/img/storage-settings-ecs-fargate-efs.png)

>     5. You can configure the Environment Values for the Appsmith in the Environment Section. For sensitive values, it's **recommended** you create secrets and set the `env` value using the ValueFrom option by specifying the `arn` of the secret created.

Reviewer: ```suggestion
    5. You can configure the environment variable for Appsmith in the **Environment** Section. For sensitive values, it's **recommended** that you create secrets and set the `env` value using the **ValueFrom** option by specifying the `arn` of the secret created.
```

Reviewer 2: @sum35h - Can we rename Environment Values to environment variables? I presume this is on AWS.
